Kernel/Syscalls/thread.cpp

```diff
@@ -208,19 +208,19 @@ ErrorOr<FlatPtr> Process::sys$get_thread_name(pid_t tid, Userspace<char*> buffer
     if (!thread || thread->pid() != pid())
         return ESRCH;
 
-    SpinlockLocker locker(thread->get_lock());
-    auto thread_name = thread->name();
+    TRY(thread->name().with([&](auto& thread_name) -> ErrorOr<void> {
+        if (thread_name->view().is_null()) {
+            char null_terminator = '\0';
+            TRY(copy_to_user(buffer, &null_terminator, sizeof(null_terminator)));
```

Comment:
This is dangerous. We cannot safely `copy_to_user` while acquiring a spinlock, because if we would page-fault, no interrupt could happen in this scope. we must copy it to another safe location and then when we are outside the spinlock scope, copy it to the user.

Reply:
I suggest doing it like this:
```c++
auto thread_name = TRY(thread->name().with([&](auto& thread_name) -> ErrorOr<NonnullOwnPtr<KString>> {
    ... your great code here to allocate a KString if needed ...
}));
TRY(copy_to_user(buffer, thread_name->view().characters_without_null_termination(), thread_name->length() + 1));
```

Reply:
It's fine to use `copy_to_user` while holding a spinlock. If a write fault occurs, it will short-circuit via the "safemem" mechanism and return `EFAULT`.

Reply:
Andreas explained to me in the Discord server that while we are in the spinlock scope, disabling interrupts doesn't mean we disable exceptions too, so please dismiss my comment entirely :)

Reply:
Thanks for raising this anyway, I still appreciate it. :^)